#!/bin/sh /usr/share/dpatch/dpatch-run
## 303-magic-add-apple2.dpatch by Radek Vokal <rvokal@redhat.com>
##
## DP: Adds new magic for Apple Emulator 2IMG format.

@DPATCH@

diff -Naur file-4.20.orig/magic/Magdir/apple file-4.20/magic/Magdir/apple
--- file-4.20.orig/magic/Magdir/apple	2006-03-02 22:10:26.000000000 +0000
+++ file-4.20/magic/Magdir/apple	2007-04-03 14:03:21.000000000 +0000
@@ -10,6 +10,23 @@
 0	belong		0x00051600		AppleSingle encoded Macintosh file
 0	belong		0x00051607		AppleDouble encoded Macintosh file
 
+# Type: Apple Emulator 2IMG format
+# From: Radek Vokal <rvokal@redhat.com>
+0	string		2IMG	Apple ][ 2IMG Disk Image
+>4	string		XGS!	\b, XGS
+>4	string		CTKG	\b, Catakig
+>4	string		ShIm	\b, Sheppy's ImageMaker
+>4	string		WOOF	\b, Sweet 16
+>4	string		B2TR	\b, Bernie ][ the Rescue
+>4	string		!nfc	\b, ASIMOV2
+>4	string		x	\b, Unknown Format
+>0xc	byte		00	\b, DOS 3.3 sector order
+>>0x10	byte		00	\b, Volume 254
+>>0x10	byte&0x7f	x	\b, Volume %u
+>0xc	byte		01	\b, ProDOS sector order
+>>0x14	short		x	\b, %u Blocks
+>0xc	byte		02	\b, NIB data
+
 # magic for Newton PDA package formats
 # from Ruda Moura <ruda@helllabs.org>
 0	string	package0	Newton package, NOS 1.x,
